### ISSUE TITLE
Change NMI and ARI to use scib library instead of sklearn

### ISSUE THREAD

**Issue by chouellaine** [chanzuckerberg/cz-benchmarks#254]
To support additional ways of computing the NMI and ARI metrics, change the library import in metrics.implementation.py from sklearn.metrics.adjusted_rand_score and sklearn.metrics.normalized_mutual_info_score to scib.metrics import nmi, ari

for v0.9 release, we have confirmed that this change does not significantly impact the metric values, so no action is needed for v0.1 and this feature can be supported after v0.1

```
from scib.metrics import nmi, ari
from czbenchmarks.tasks import ClusteringTask
from czbenchmarks.tasks.constants import RANDOM_SEED, FLAVOR, KEY_ADDED, OBSM_KEY

obsm_key = OBSM_KEY
random_seed = RANDOM_SEED
n_iterations = 2
flavor = FLAVOR
key_added = KEY_ADDED
tissues = ["salivary_gland","mammary","testis"]
czb_results = list()
scib_results = list()
for tissue in tissues:
    with open(f'tsv2_{tissue}_GENEFORMER_gf_12L_95M_processed.dill', 'rb') as f:
        gf = dill.load(f)

    with open(f'tsv2_{tissue}_UCE_4l_processed.dill', 'rb') as f:
        uce4 = dill.load(f)

    with open(f'tsv2_{tissue}_UCE_33l_processed.dill', 'rb') as f:
        uce33 = dill.load(f)
    for model_name, dataset in [("GF",gf), ("UCE4",uce4), ("UCE33",uce33)]:
        task = ClusteringTask("cell_type")
        result = task.run(dataset)
        czb_results.append({"model":model_name,"tissue":tissue,"result":list(result.values())})
        adata = dataset.adata
        sc.pp.neighbors(adata, use_rep=obsm_key, random_state=random_seed)
        sc.tl.leiden(
            adata,
            key_added=key_added,
            flavor=flavor,
            n_iterations=n_iterations,
            random_state=random_seed,
        )
        adata.obs["cluster"] = adata.obs[key_added]
        nmi_result = nmi(adata, cluster_key="cluster", label_key="cell_type")
        ari_result = ari(adata, cluster_key="cluster", label_key="cell_type")
        scib_results.append({"model":model_name,"tissue":tissue,"metric":"SCIB_NMI", "value":nmi_result})
        scib_results.append({"model":model_name,"tissue":tissue,"metric":"SCIB_ARI", "value":ari_result})

df = pd.DataFrame(czb_results)
df = df.explode("result").explode("result")
df["metric"] = df["result"].apply(lambda x: x.metric_type.name)
df["value"] = df["result"].apply(lambda x: x.value)
df["params"] = df["result"].apply(lambda x: x.params)
df = df.drop(columns=["result"])

df_scib = pd.DataFrame(scib_results)
final = pd.concat([df, df_scib]).drop(columns="params")
final = pd.pivot(final, index=["model","tissue"], columns="metric", values="value").reset_index()
final["ari_delta"] = final["SCIB_ARI"] - final["ADJUSTED_RAND_INDEX"]
final["nmi_delta"] = final["SCIB_NMI"] - final["NORMALIZED_MUTUAL_INFO"]
```

**Comment by chouellaine** [comment]
<img width="1213" alt="Image" src="https://github.com/user-attachments/assets/0e7ff8ac-294a-4583-88be-a70b5b008f48" />
